### ISSUE TITLE
Support for Depth Backward Propagation in Rasterizer?

### ISSUE THREAD

**Issue by WWzzEE2** [nerfstudio-project/gsplat#565]
Thank you for your excellent work on this project! The significant improvement in training speed for 3DGS is truly impressive.

I noticed that the rasterizer supports depth rendering. Could you please clarify if it currently supports depth backward propagation? I am interested in incorporating depth supervision during training and would like to understand the feasibility.

**Comment by abrahamezzeddine** [comment]
I see this in the trainer;

# Enable depth loss. (experimental)
depth_loss: bool = False

self.trainset = Dataset(
    self.parser,
    split="train",
    patch_size=cfg.patch_size,
    load_depths=cfg.depth_loss,  # Loads depth maps if enabled
)

renders, alphas, info = self.rasterize_splats(
    # ...
    render_mode="RGB+ED" if cfg.depth_loss else "RGB",
    # ...
)
if renders.shape[-1] == 4:
    colors, depths = renders[..., 0:3], renders[..., 3:4]  # Split into RGB and depth
else:
    colors, depths = renders, None

if cfg.depth_loss:
    # Sample depth predictions at specific points
    points = torch.stack([
        points[:, :, 0] / (width - 1) * 2 - 1,
        points[:, :, 1] / (height - 1) * 2 - 1,
    ], dim=-1)
    grid = points.unsqueeze(2)  # [1, M, 1, 2]
    depths = F.grid_sample(
        depths.permute(0, 3, 1, 2), grid, align_corners=True
    )
    depths = depths.squeeze(3).squeeze(1)  # [1, M]
    
    # Calculate loss in disparity space (1/depth)
    disp = torch.where(depths > 0.0, 1.0 / depths, torch.zeros_like(depths))
    disp_gt = 1.0 / depths_gt  # [1, M]
    depthloss = F.l1_loss(disp, disp_gt) * self.scene_scale
    loss += depthloss * cfg.depth_lambda

Could this help you?

I would also be happy to try out your solution if it works!

**Comment by WWzzEE2** [comment]
I have conducted tests using my own depth masks as supervision, and GSplat performs as expected. It seems that the library already supports depth propagation. Thank you for your contributions and feedback!
